metcalcpy/util/ecnt_statistics.py

```diff
@@ -134,18 +112,53 @@ def calculate_ecnt_crps_emp_fair(input_data, columns_names, aggregation=False):
 
         Returns:
             calculated ECNT_CRPS_EMP_FAIR as float
-            or None if some of the data values are missing or invalid
+            or None if some data values are missing or invalid
+    """
+
+    return weighted_average(input_data, columns_names, 'crps_emp_fair', aggregation)
+
+
+def calculate_ecnt_spread_md(input_data, columns_names, aggregation=False):
+    """Performs calculation of SPREAD_MD - The pairwise Mean Absolute Difference
+        of the unperturbed ensemble members
+
+        Args:
+            input_data: 2-dimensional numpy array with data for the calculation
+                1st dimension - the row of data frame
+                2nd dimension - the column of data frame
+            columns_names: names of the columns for the 2nd dimension as Numpy array
+            aggregation: if the aggregation on fields was performed
+
+        Returns:
+            calculated SPREAD_MD as float
+            or None if some data values are missing or invalid
+    """
+    return weighted_average(input_data, columns_names, 'spread_md', aggregation)
+
+
+def weighted_average(input_data, columns_names, column_name, aggregation=False):
+    """ Performs aggregation over multiple cases using a weighted average approach,
+     where the weight is defined by the number of matched pairs in the TOTAL column
+
+    :param input_data: 2-dimensional numpy array with data for the calculation
+                1st dimension - the row of data frame
+                2nd dimension - the column of data frame
+    :param columns_names: names of the columns for the 2nd dimension as Numpy array
+    :param column_name: name of the column to be aggregated
+    :param aggregation: if the aggregation on fields was performed
+    :return: aggregated column values or None if some data values are missing or invalid
     """
     warnings.filterwarnings('error')
     try:
         total = get_total_values(input_data, columns_names, aggregation)
-        crps_emp_fair = sum_column_data_by_name(input_data, columns_names, 'crps_emp_fair') / total
-        result = round_half_up(crps_emp_fair, PRECISION)
+        statistic = sum_column_data_by_name(input_data, columns_names, column_name) / total
```

Comment:
@TatianaBurek, yes, good point I suspect there's an issue here as well. It's possible I just don't understand how this code is working. But it looks to me like you're summing up the STATISTIC values and TOTAL counts and dividing the first by the second.

Using R to provide an example of aggregating MAE values, where the weight is defined by the total column:
```
R
MAE = c(5, 10, 8)
TOTAL = c(1000, 1500, 1250)
c("Bad aggregated value using this logic = ", sum(MAE) / sum(TOTAL))
# Prints incorrect value of 0.00613
c("Correct weighted aggregation = ", sum(MAE*TOTAL)/sum(TOTAL))
# Prints correct value of 8
```
Those values are so different, so I'm assuming this IS working, but I'm just not grasping how.

Reply:
the preprocessing of the data  - calculation of the additional statistics , multiplying  by the weight, renaming columns - is happening in agg_stat.py script. For example, here I prepare ecnt data:
https://github.com/dtcenter/METcalcpy/blob/deedf36fc910aee5b970e9fcfb2de199e771bc14/metcalcpy/agg_stat.py#L448
calculate_<statistic> methods use the data that was already multiplied  by the weight

Reply:
The decision on separating preparing and calculating data was made based on the need to improve the speed of bootstraping process. It makes more sense to precalculate base values once and use these values n-replication times during  bootstraping